src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -245,13 +245,15 @@ public void removeAll() {
      */
     @Override
     public HashSet<String> getUniqueUsersWithTokenCache() {
-        Iterator<TokenCacheItem> results = this.getAll();
+        Iterable<TokenCacheItem> results = this.getAll();
         HashSet<String> users = new HashSet<String>();
-
-        while (results.hasNext()) {
-            TokenCacheItem item = results.next();
-            if (item.getUserInfo() != null && !users.contains(item.getUserInfo().getUserId())) {
-                users.add(item.getUserInfo().getUserId());
+        
+        if (results != null) {
```

Comment:
Why do you always check it for null, your implementation of getAll() will never return null, if null is unexpected don't check for null